### ISSUE TITLE
Alpha option doesn't work well with logarithmic parameters

### ISSUE THREAD

**Issue by rodluger** [rodluger/vplot#2]
Running ``vplot -x Temp -y Lum -a Time`` in ``rodrigo/stellar`` produces the following:

![tmp](https://cloud.githubusercontent.com/assets/9323819/10093993/102d8b92-630f-11e5-9073-57aa0e51ddd1.jpg)

Note that alpha isn't changing logarithmically with time (increases suddenly from 0.1 to 1 only at the bottom tip of each curve). The dashes are also screwed up on the ``0.6MSUN`` and ``0.7MSUN`` curves.

**Comment by rodluger** [comment]
Mostly resolved; alpha now scales logarithmically based on the ``alog`` parameter. Dashed lines still don't plot well, though, depending on the resolution of the line.

**Comment by deitrr** [comment]
How is the alpha supposed to look? I am having trouble visualizing it.

Also, I have had trouble in the past with dashed lines, but mostly with
data that isn't smooth. Have you tried dotted lines?

**Comment by rodluger** [comment]
In this plot, alpha increases logarithmically with time:
![tmp](https://cloud.githubusercontent.com/assets/9323819/10105870/0501ef72-6366-11e5-971d-1bce58e4cb14.png)

Dotted lines are a good idea! I'll update the defaults to that.

**Comment by deitrr** [comment]
Yeah, that looks much better!

Dashed lines plot weird when you have lots of small curvature in your data,
because the dashes and spaces all bend to keep up with that curvature, but
the length of each dash and space stays fixed. Looking at how the spaces
and dashes seem to have varying sizes, I'm guessing that is what is going
on in your data. If the data is dense you could try decimating it to make
it smoother (e.g., plot data[::10]). It's annoying that this happens,
though.

**Comment by rodluger** [comment]
Yeah. The issue is that there's no way to add varying alpha to a single curve in matplotlib (as far as I know). I had to create a LineCollection, which is several lines stitched together, each with its own alpha and line style. If the segment is smaller than the spacing between the dashes, it will simply appear solid; several small segments next to each other therefore create a solid line, even if their individual styles are dashed. I tried plotting every *nth* point, and it kind of works, but it reduces the line density everywhere; some parts of the curve that have few points to begin with get really undersampled.

**Comment by deitrr** [comment]
Oh man, that is tricky.
